Slingshot.Api/Concrete/BitbucketRepository.cs

```diff
@@ -302,14 +309,28 @@ public BitbucketRepository(Uri uri, string host, string token)
             }
         }
 
-        public class OAuthInfo
+        /// <summary>
+        /// Partial object, full reponse please reference
+        /// https://confluence.atlassian.com/bitbucket/pullrequests-resource-423626332.html#pullrequestsResource-GETaspecificpullrequest
+        /// </summary>
+        public class BitBucketPullRequest : IPullRequestInfo
+        {
+            public JObject source { get; set; }
+
+            public string SourceBranch
+            {
+                get
+                {
+                    return this.source["branch"]["name"].Value<string>();
```

Comment:
Might need to catch exception and rethrow a cleaner exception that says you couldn't find a branch in the payload. Otherwise, we'll just get a strange null ref exception.